logic/gateway.go

```diff
@@ -112,6 +112,9 @@ func CreateIngressGateway(netid string, nodeid string, ingress models.IngressReq
 	if host.FirewallInUse == models.FIREWALL_NONE {
 		return models.Node{}, errors.New("firewall is not supported for ingress gateways")
 	}
+	if host.NatType != models.NAT_Types.Public {
+		return models.Node{}, errors.New("ingress cannot be created on nodes behind NAT")
+	}
```

Comment:
why not allow it?
clients can still connect over the public listen port right?

Reply:
Whats a "public listen port" if not a port **not** behind NAT (called "public NAT" in this case).

Reply:
mentioned by @TobiaszCudnik if there is port forwarding, it should be detected as public anyway